### DISCUSSION TITLE
Irmin as a PouchDB/CouchDB replacement?

### DISCUSSION THREAD

**Discussion by hoichi** [mirage/irmin#1282] [Q&A] [answered]
Hey guys!

Hope it's ok to put questions like these in an issue, I've found no other means of communication.

I work in a team using OCaml/ReScript, and we're considering Irmin as a solution for collaborative editing. We want (maybe to different degrees): collaborative editing, version history, and two-way replication between browsers and the server.

So my question is: how feasible it is to use Irmin in the browser today. Is it production-ready? And more specifically:
- Do you know if you can compile Irmin with ReScrit(formerly BuckleScript)? I've only seen mentions of JSOO.
- ~~Is there a ready-made backend for LocalStorage/IndexedDB, or do we have to write it ourselves?~~ Nevermind, found [talex5/irmin-indexeddb](https://github.com/talex5/irmin-indexeddb)
- What about two-way replication between the browser and the server? Will we have to write the replication layer too?

**Comment by avsm** [comment, marked as answer]
@hoichi thanks for the interest. I'd suggest short-circuiting a back and forth on this issue to have a quick video chat to discuss what you'd like to do, and then circle back and summarise the findings in this issue.  Browser support has long been on our roadmap, and we've had prototypes, but it's been awaiting a serious user before committing to it. And here you are :-)

If you could drop me a mail on anil@recoil.org, I'll connect to the right Irmin/Mirage developers.

**Reply by hoichi** [reply]
Hey Anil! Thanks for being ready to help. Unfortunately, a few hours after I've posted the question, we decided to put the project on pause. We need to figure out our priorities first. But, if we decide to move on with it, I'll be sure to contact you. Thanks again, and sorry for the noise!

**Comment by craigfe** [comment]
I've migrated this issue to a "Discussions" thread, which will hopefully be a better environment for whatever follow-up discussion we have :-)

**Reply by hoichi** [reply]
I didn't know discussions even existed :facepalm:

**Comment by gdamjan** [comment]
so, a year and a half later, are there any updates on this topic? :)

I'm not a serious user though, I just wanna play with irmin for a toy rss reader I want to build (and server client synchronization).

**Reply by hoichi** [reply]
Not from me :) Not only haven’t the project moved on during my time, I’ve left the team about half a year ago.

**Reply by zshipko** [reply]
You could look into using [irmin-server](https://github.com/mirage/irmin-server) with irmin-client-jsoo - this doesn’t persist data in the browser but it gets you most of the way there.

Here’s the example browser client from the repo: https://github.com/mirage/irmin-server/tree/master/examples/js
